services/customerService.js

```diff
@@ -0,0 +1,66 @@
+const yup = require('yup');
+const customerModel = require('../models/customerModel');
+
+const getAll = async (req, res, next) => {
+  let data;
+  try {
+    data = await customerModel.getAll();
+    res.status(200).send(data);
+  } catch (e) {
+    console.log(e);
+    next(e);
+  }
+};
+
+const getCustomerById = async (req, res, next) => {
+  let data;
+  try {
+    data = await customerModel.getCustomerById(req.params.id);
+    res.status(200).send(data);
+  } catch (e) {
+    console.log(e);
+    next(e);
+  }
+};
+
+const create = async (req, res, next) => {
+  const customer = {
+    ...req.body,
+  };
+
+  // check to remove empty string
+  if (!customer.birthdate) {
+    customer.birthdate = undefined;
+  }
+
+  // validation of schema for necessary fields
+  const schema = yup.object().shape({
+    first_name: yup.string().required(),
+    last_name: yup.string().required(),
+    address: yup.string(),
+    city: yup.string(),
+    state: yup.string(),
+    zip: yup.string(),
+    image: yup.string(),
+    birthdate: yup.date().nullable().default(null),
+  });
+
+  schema
+    .validate(customer)
+    .then(async (validatedCustomer) => {
+      const savedCustomer = await customerModel.createCustomer(
```

Comment:
I'm still a little unfamiliar with this syntax. What is returned and assigned to `savedCustomer` in this case?

Reply:
The result here:

```
  const result = await db.pool.asyncQuery(
     `SELECT * FROM Customers WHERE id = (SELECT MAX(id) FROM Customers)`,
   );
   return result[0];
```